### DISCUSSION TITLE
Pre-select a shipping or payment method at checkout

### DISCUSSION THREAD

**Discussion by SaeedYasin** [Shopify/function-examples#237] [API Ideas]
It would be nice if using functions we can pre-select a certain shipping or payment method. Based on the order details, store owners may prefer to show a pre-selected option at checkout which would be the best choice for that order. As many customers don't bother with changing the pre-selected option and get into issues later with their order.

**Comment by gil--** [comment]
Just wondering, does using `moveoperation` not solve this since you can move an option to index 0 which causes it to be auto selected?https://shopify.dev/docs/api/functions/reference/delivery-customization/graphql/common-objects/moveoperation

**Reply by SaeedYasin** [reply]
Yes, but it would be nice to have better control over it.

**Reply by blanklob** [reply]
it deosn't work when you are in checkout already.

**Reply by sebastyuiop** [reply]
I'm guessing the move operation would move the selected shipping method for all customers? We'd like to be able to pre-select the option based on our own logic on a per customer basis.

**Comment by SaeedYasin** [comment]
There are stores on which moving delivery or payment option using the moveoperation to index 0 location, does not cause it to be auto selected. For example, `happyhoundlife.myshopify.com` which had payment customization which will reorder payment methods and the payment methods do get sorted, but index zero payment method is not pre-selected.

This is not a cache issue, opening the store in a new browser and creating a new checkout still always selects Klarna as the pre-selected payment method, which is shown at the last index. How is this possible that index 0 payment method is not the default pre-selected option and what can we do about it?

**Comment by rbarreca** [comment]
Also, I'd expand this feature request to be able to both pre-select "Ship" vs "Pick up" (to expand those option groups) as well as pre-select the shipping method or pickup location. ([Original comment here](https://github.com/Shopify/function-examples/discussions/218#discussioncomment-8028286).)

**Reply by blanklob** [reply]
Yes this please!

**Comment by dani-sanomads** [comment]
Are we expecting updates for this pre-selection of delivery / payment methods? @rbarreca

**Comment by blanklob** [comment]
This is actually kinda released on last editions but only works with sf api and headless cart create mutation.

**Comment by Waynenabors** [comment]
@gil-- @blanklob thanks for the info ☝️ sounds like there is still no way to set pickup as the preffered delivery method without the Storefront API. We have a liquid storefront and were hoping customer could choose from the product page

**Comment by blanklob** [comment]
@Waynenabors i agree, we need this on Ajax card and as form attribute. all major storfront do this logic in cart or even in PDP.

**Comment by sebastyuiop** [comment]
We also need this on the Ajax cart. Ideally we could pass in the selected shipping method so that it is pre-selected at checkout. Thank you

**Comment by simonlayfield** [comment]
\+  1 for this
